frontend/src/views/CellGuide/components/common/OntologyDagView/components/Legend/index.tsx

```diff
@@ -1,25 +1,144 @@
+  const targetNodeLegendComponent = (
+    <LegendItemWrapper>
+      Cell Type
+      <IsTargetNode />
+      Selected
+    </LegendItemWrapper>
+  );
+  const size = largeSize * 2;
+  const descendantsLegendComponent = (
+    <LegendItemWrapper>
+      <FlexColumn>
+        <NodeWrapper>
+          <NoDescendantsLine size={size} />
+          <StyledBorder width={1} height={size * 2} />
+        </NodeWrapper>
+        No Descendants
+      </FlexColumn>
+    </LegendItemWrapper>
+  );
+  const corpusLegendComponent = (
+    <>
+      <LegendItemWrapper>
+        <LegendItem>
+          <FlexColumn>
+            <IsInCorpus />
+            <CenterText>In Census</CenterText>
+          </FlexColumn>
+        </LegendItem>
+      </LegendItemWrapper>
+      <LegendItemWrapper>
+        <LegendItem>
+          <FlexColumn>
+            <IsNotInCorpus />
+            <CenterText>Not In Census</CenterText>
+          </FlexColumn>
+        </LegendItem>
+      </LegendItemWrapper>
+    </>
+  );
+
+  const numPies = 6;
+  const numMarkerScores = maxMarkerScore + 1;
+
+  const expressedInCellsPercLegendComponent = (
+    <LegendItemWrapper>
+      Expressed in Cells(%)
+      <MarkerScoreWrapper>
+        {Array.from({ length: numPies }).map((_, i) => (
+          <FlexColumn key={i}>
+            <ExpressedInCells
+              degree={(360 / (numPies - 1)) * i}
+              fill={secondaryColor}
+              size={size}
+            />
+            {(i == 0 || i == numPies - 1) && (
+              <CenterText>{(100 / (numPies - 1)) * i}</CenterText>
+            )}
+          </FlexColumn>
+        ))}
+      </MarkerScoreWrapper>
+    </LegendItemWrapper>
+  );
+
+  const markerScoreLegendComponent = (
+    <LegendItemWrapper>
+      Marker Score
+      <MarkerScoreWrapper>
+        {Array.from({ length: numMarkerScores }).map((_, i) => (
+          <FlexColumn key={i}>
+            <MarkerScore fill={i} size={size} />
+            {(i == 0 || i == numMarkerScores - 1) && (
+              <CenterText>{i}</CenterText>
+            )}
+          </FlexColumn>
+        ))}
+      </MarkerScoreWrapper>
+    </LegendItemWrapper>
+  );
+  const hasMarkerGeneLegend = (
+    <>
+      {markerScoreLegendComponent}
+      {expressedInCellsPercLegendComponent}
+    </>
+  );
   return (
-    <g>
-      <rect
-        x={width - 260}
-        y={-10}
-        width={260}
-        height={60}
-        fill={backgroundColor}
-        rx={4}
-      />
-      <InCorpusLegend xPos={width - 240} yPos={10} />
-      <DescendantsLegend xPos={width - 160} yPos={10} />
-      <CollapsedNodesLegend xPos={width - 80} yPos={10} />
-    </g>
+    <LegendWrapper data-testid={CELL_GUIDE_ONTOLOGY_VIEW_LEGEND_TEST_ID}>
+      {!selectedGene && !isTissue && targetNodeLegendComponent}
+      {selectedGene ? hasMarkerGeneLegend : corpusLegendComponent}
+      {descendantsLegendComponent}
+    </LegendWrapper>
   );
 }
+
+interface ExpressedInCellsProps {
```

Comment:
Ideally it would be great if we can put `types` in a separate types.ts consistently, instead of embedding it in the code (like here: frontend/src/views/CellGuide/components/common/OntologyDagView/common/types.ts)